docs/rule/no-passed-in-event-handlers.md

```diff
@@ -0,0 +1,43 @@
+## no-passed-in-event-handlers
+
+It is possible to pass e.g. `@click` to an Ember component to override the
+default `click` event handler. For tagless components this will trigger an
+assertion though and can't be used as legitimate API, and for Glimmer
+components it will not work out of the box, like in Ember components, either.
+
+This rule scans potential component invocations for these patterns and flags
+them as issues.
+
+### Examples
+
+This rule **forbids** the following:
+
+```hbs
+<Foo @click={{this.handleClick}} />
+```
+
+```hbs
+<Foo @keyPress={{this.handleClick}} />
+```
+
+```hbs
+{{foo click=this.handleClick}}
+```
+
+This rule **allows** the following:
+
+```hbs
+<Foo @onClick={{this.handleClick}} />
```

Comment:
I'm worried that this example could make `onClick` seem like a real Ember API. It may help to either add a comment explaining that it's a custom name or use a name like `myCustomClickHandler={{this.handleClick}}`.

Reply:
good point, I've added your `myCustomClickHandler` example